src/util/atexit.go

```diff
@@ -0,0 +1,45 @@
+package util
+
+import (
+	"os"
+	"sync"
+)
+
+var atExitFuncs struct {
+	fns []func()
+	mu  sync.Mutex
+}
+
+// AtExit registers the function fn to be called on program termination.
+// The functions will be called in reverse order they were registered.
+func AtExit(fn func()) {
+	if fn == nil {
+		panic("AtExit called with nil func")
+	}
+	atExitFuncs.mu.Lock()
+	defer atExitFuncs.mu.Unlock()
+	once := &sync.Once{}
+	atExitFuncs.fns = append(atExitFuncs.fns, func() {
+		once.Do(fn)
+	})
+}
```

Comment:
Please correct me if I'm wrong, but I don't think we need a mutex here, unless we're exporting the code for other programs, which fzf isn't doing.

Reply:
Good catch, I removed the mutexes.